lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -886,6 +913,46 @@ impl SublayersModel {
 
 
 
+// ===================
+// === MaskedLayer ===
+// ===================
+
+/// A layer with an attached mask. Each opaque shape in the `mask_layer` defines the renderable area
+/// of the `masked_layer`.
```

Comment:
wait, is our masking system working based on opaque/non-opaque shapes? Hmm, we should change it and allow two modes - either this or white/black masks. Please do not describe in this comment how making works (opaque vs black/white) - instead, redirect it to docs of masking impl.

Reply:
Yes, it is. Done.